### ISSUE TITLE
markdown incorrectly handles literal backticks in code spans

### ISSUE THREAD

**Issue by jbristow** [thepracticaldev/dev.to#1284]
**Describe the bug**
When creating code spans inside regular text, adding backticks seems to break formatting.

**To Reproduce**
The following markdown:
````markdown
Hello, I am `` `__` `` and I am happy to meet you!

My friend `` `foo` `` is also here!

Watch out! `` `bar`.explosion() ``

Oh no! `` chunk(`baz`) ``

And now for some more code!
```prolog
parent(charles1, james1).
parent(elizabeth, james1).
```
Some closing remarks.
````

**Expected behavior**
I was expecting one of these:
https://github.github.com/gfm/#code-spans
https://daringfireball.net/projects/markdown/syntax#code

**Screenshots**
<img width="497" alt="screen shot 2018-12-07 at 2 12 11 pm" src="https://user-images.githubusercontent.com/314260/49675356-23250800-fa2a-11e8-966f-d046a51cc87b.png">
(Note the improperly highlighted initial/final whitespace)

**Desktop (please complete the following information):**

- OS: MacOS High Sierra
- Browser: Firefox Quantum
- Version: 63.0.3 (64-bit)

**Comment by jbristow** [comment]
Recursively markdowning the markdown is similarly messed up:

`````markdown
````markdown
Hello, I am `` `__` `` and I am happy to meet you!

My friend `` `foo` `` is also here!

Watch out! `` `bar`.explosion() ``

Oh no! `` chunk(`baz`) ``

And now for some more code!
```prolog
parent(charles1, james1).
parent(elizabeth, james1).
```

Some closing remarks.
````
`````

<img width="479" alt="screen shot 2018-12-07 at 2 17 45 pm" src="https://user-images.githubusercontent.com/314260/49675522-e9a0cc80-fa2a-11e8-9b63-6460e3750b69.png">

**Comment by Zhao-Andy** [comment]
Hey @jbristow, thanks for the report. This should be fixed now. Let us know if otherwise!

**Comment by achasveachas** [comment]
@Zhao-Andy it doesn't look like this was fixed.

I'm writing a blog post with markdown tips so naturally I would like to include examples that inclde backticks and none of the recommended ways of displaying backticks seem to work.

Specifically, trying to render three backticks in inline code seems to be broken:
```` Something like: (`` ``` ``)```` I would expect to render like: Something like (`` ``` ``) but in the DEV editor it comes out like: Something like (` ` `).

Similarly when trying to display three backticks insode a code block things break down.

The following:
`````
````
```
//some code goes here
```
````
`````

**Comment by achasveachas** [comment]
@Zhao-Andy it doesn't look like this was fixed.

I'm writing a blog post with markdown tips so naturally I would like to include examples that inclde backticks and none of the recommended ways of displaying backticks seem to work.

Specifically, trying to render three backticks in inline code seems to be broken:
```` Something like: (`` ``` ``)```` I would expect to render like: Something like (`` ``` ``) but in the DEV editor it comes out like: Something like (` ` `).

Similarly when trying to display three backticks insode a code block things break down.

The following markdown:
`````
````
```
//some code goes here
```
````
`````
I'd expect to render like:

````
```
//some code goes here
```
````

But instead it renders on DEV like:
```
```
//some code goes here
```
```

**Comment by jbristow** [comment]
Agreed, while my first set of examples renders properly, my second set (the double escaped markdown) does not render properly.

I swear that it was fine at the time of closing, so either this is a regression, or I didn’t check as thoroughly as I remember.

**Comment by Zhao-Andy** [comment]
Thanks for the heads up! We'll take a look into it again.

**Comment by achasveachas** [comment]
I don't know if there's a need to reopen this issue, I opened https://github.com/thepracticaldev/dev.to/issues/3648
